util/mod_translation_updater.py

```diff
@@ -133,6 +133,19 @@ def main():
 	r'\[\[(.*?)\]\]' # [[ ... ]] string delimiters
 	r'[\s,\)]', # Same as for pattern_lua_quoted
 	re.DOTALL)
+# Handles like pattern_lua_quoted, but for single parameter (without parantheses)
+# See https://www.lua.org/pil/5.html for informations about single argument call
+pattern_lua_quoted_single = re.compile(
+	r'(?:^|[\.=,{\(\s])' # Same as for pattern_lua_quoted
```

Comment:
If you have these "same" parts, why not just extract them into variables with useful names?

Reply:
> If you have these "same" parts, why not just extract them into variables with useful names?

Something like this?

<details>
<summary>Python code</summary>

```python
# List of pattern components:
# Look for beginning of file or anything that isn't a function identifier
pattern_lua_begin_non_function = r'(?:^|[\.=,{\(\s])'
# Matches S, FS, NS, or NFS function call
pattern_lua_NFS_function = r'N?F?S\s*'
# Start of argument: open parentheses and space (optional)
pattern_lua_open_parentheses = r'\(\s*'
# Quoted string
# Group 2 will be the string, group 1 and group 3 will be the delimiters (" or ')
# See https://stackoverflow.com/questions/46967465/regex-match-text-in-either-single-or-double-quote
pattern_lua_quoted_string = r'(["\'])((?:\\\1|(?:(?!\1)).)*)(\1)'
# Double square bracket string (multiline)
pattern_lua_square_bracket_string = r'\[\[(.*?)\]\]'
# End of argument or function call: space, comma, or close parentheses
pattern_lua_close_parentheses = r'[\s,\)]'

# Handles the " ... " or ' ... ' string delimiters
pattern_lua_quoted = re.compile(
	pattern_lua_begin_non_function +
	pattern_lua_NFS_function +
	pattern_lua_open_parentheses +
	pattern_lua_quoted_string +
	pattern_lua_close_parentheses,
	re.DOTALL)
# Handles the [[ ... ]] string delimiters
pattern_lua_bracketed = re.compile(
	pattern_lua_begin_non_function +
	pattern_lua_NFS_function +
	pattern_lua_open_parentheses +
	pattern_lua_square_bracket_string +
	pattern_lua_close_parentheses,
	re.DOTALL)
# Handles like pattern_lua_quoted, but for single parameter (without parentheses)
# See https://www.lua.org/pil/5.html for informations about single argument call
pattern_lua_quoted_single = re.compile(
	pattern_lua_begin_non_function +
	pattern_lua_NFS_function +
	pattern_lua_quoted_string,
	re.DOTALL)
# Same as pattern_lua_quoted_single, but for [[ ... ]] string delimiters
pattern_lua_bracketed_single = re.compile(
	pattern_lua_begin_non_function +
	pattern_lua_NFS_function +
	pattern_lua_square_bracket_string,
	re.DOTALL)
```
</details>

> Also, I see no reason why this can't be a single regex. Everything is common except for the "body", so you could just do ... common stuff ... (quoted string|bracketed string), right?

(See my other reply.)

> (Furthermore, I don't see why this wouldn't be able to share a pattern with parenthesized function calls even.)

I have not checked that.

Reply:
Based on [the discussion over #minetest-dev](https://irc.minetest.net/minetest-dev/2024-05-12#i_6170441) with @appgurueu, this is the revised code:

<details>
<summary>Python code</summary>

```python
# Compile pattern for matching lua function call
def compile_func_call_pattern(argument_pattern):
	return re.compile(
		# Look for beginning of file or anything that isn't a function identifier
		r'(?:^|[\.=,{\(\s])' +
		# Matches S, FS, NS, or NFS function call
		r'N?F?S\s*' +
		# The pattern to match argument
		argument_pattern,
		re.DOTALL)

# Add parentheses around a pattern
def parenthesize_pattern(pattern):
	return (
		# Start of argument: open parentheses and space (optional)
		r'\(\s*' +
		# The pattern to be parenthesized
		pattern +
		# End of argument or function call: space, comma, or close parentheses
		r'[\s,\)]')

# Quoted string
# Group 2 will be the string, group 1 and group 3 will be the delimiters (" or ')
# See https://stackoverflow.com/questions/46967465/regex-match-text-in-either-single-or-double-quote
pattern_lua_quoted_string = r'(["\'])((?:\\\1|(?:(?!\1)).)*)(\1)'

# Double square bracket string (multiline)
pattern_lua_square_bracket_string = r'\[\[(.*?)\]\]'

# Handles the " ... " or ' ... ' string delimiters
pattern_lua_quoted = compile_func_call_pattern(parenthesize_pattern(pattern_lua_quoted_string))

# Handles the [[ ... ]] string delimiters
pattern_lua_bracketed = compile_func_call_pattern(parenthesize_pattern(pattern_lua_square_bracket_string))

# Handles like pattern_lua_quoted, but for single parameter (without parentheses)
# See https://www.lua.org/pil/5.html for informations about single argument call
pattern_lua_quoted_single = compile_func_call_pattern(pattern_lua_quoted_string)

# Same as pattern_lua_quoted_single, but for [[ ... ]] string delimiters
pattern_lua_bracketed_single = compile_func_call_pattern(pattern_lua_square_bracket_string)
```
</details>